### PR TITLE
Remove Lines Containing String

### PR DESCRIPTION
**Pull request by vnkarthik87**
Added functionality to remove lines matching the string pattern in
String.py file.
This keyword seems more required when one wants to get string lines that doesn't contain given pattern which is Remove Lines Containing String

### CODE REVIEW

**Comment by robotci** [comment]
Admin, can CI be enabled?

**Comment by pekkaklarck** [comment]
1.`Remove Lines Containing String` is basically opposite of `Get Lines Containing String`, right? Sounds like a good idea to add it.

2. Should highly related `Remove Lines Matching Pattern` and `Remove Lines Matching Pattern` be added at the same time?

3. Could you submit an issue about this too? We prefer to have an issue about all enhancements even if there would also be a pull request. Easier to manage them that way.

4. Tests should be added. See `atest/README.rst` for information about the test system and existing String library tests for examples.

5. I'll add some further comments to the actual code.

File: src/robot/libraries/String.py [hunk 1]
```diff
@@ -346,6 +352,28 @@ def remove_string_using_regexp(self, string, *patterns):
...
+        The ``pattern`` is always considered to be a normal string and a
+        line matches if the ``pattern`` is found anywhere in it. By
+        default the match is case-sensitive, but setting
+        ``case_insensitive`` to any value makes it case-insensitive.
+
+        Non matching lines are returned as one string catenated back together with
+        newlines. Possible trailing newline is never returned. The
+        number of non matching lines is automatically logged.
+
+		This does exact opposite functionality of Get Lines Containing String
```
Comment (pekkaklarck) [inline src/robot/libraries/String.py]: 1. Weird indentation. 

2. Add backticks around referenced keyword names like

    `Get Lines Containing String`

  to automatically create a link to that keyword.

3. Add a note `New in Robot Framework 2.9.` at the end of the documentation.

4. You can review generated documentation by running

    doc/libraries/lib2html.py String

File: src/robot/libraries/String.py [hunk 2]
```diff
@@ -346,6 +352,28 @@ def remove_string_using_regexp(self, string, *patterns):
...
+        """
+        if case_insensitive:
+            pattern = pattern.lower()
+            contains = lambda line: pattern in line.lower()
+        else:
+            contains = lambda line: pattern in line
+        return self._get_non_matching_lines(string, contains)
+
```
Comment (pekkaklarck) [inline src/robot/libraries/String.py]: Extra empty line added here. We only have one between methods as PEP-8 recommends.

**Comment by pekkaklarck** [comment]
@vnkarthik87, are you interested to enhance this PR?

**Comment by pekkaklarck** [comment]
No comments in over a year. The PR already has conflicts. Closing.

**Pull request closed without merging**